src/Spice86.Core/Emulator/InterruptHandlers/Input/Mouse/MouseDriver.cs

```diff
@@ -0,0 +1,233 @@
+    public void RestoreRegisters() {
+        if (_savedRegisters == null) {
+            return;
+        }
+        _state.ES = _savedRegisters.Es;
+        _state.DS = _savedRegisters.Ds;
+        _state.DI = _savedRegisters.Di;
+        _state.SI = _savedRegisters.Si;
+        _state.BP = _savedRegisters.Bp;
+        _state.SP = _savedRegisters.Sp;
+        _state.BX = _savedRegisters.Bx;
+        _state.DX = _savedRegisters.Dx;
+        _state.CX = _savedRegisters.Cx;
+        _state.AX = _savedRegisters.Ax;
+    }
+
+    /// <inheritdoc />
+    public short GetDeltaXMickeys() {
+        double deltaXPixels = _mouseDevice.DeltaX * VirtualScreenWidth;
+        double deltaXMickeys = _mouseDevice.HorizontalMickeysPerPixel * deltaXPixels;
+        return (short)deltaXMickeys;
+    }
+
+    /// <inheritdoc />
+    public short GetDeltaYMickeys() {
+        double deltaYPixels = _mouseDevice.DeltaY * _vgaMode.Height;
+        double deltaYMickeys = _mouseDevice.VerticalMickeysPerPixel * deltaYPixels;
+        return (short)deltaYMickeys;
+    }
+
+    /// <inheritdoc />
+    public void Reset() {
+        _vgaMode = _vgaFunctions.GetCurrentMode();
+        SetCursorPosition(VirtualScreenWidth / 2, _vgaMode.Height / 2);
+        _mouseCursorHidden = -1;
+        _gui?.HideMouseCursor();
+        CurrentMinX = 0;
+        CurrentMinY = 0;
+        CurrentMaxX = VirtualScreenWidth - 1;
+        CurrentMaxY = _vgaMode.Height - 1;
+        HorizontalMickeysPerPixel = 8;
+        VerticalMickeysPerPixel = 16;
+        DoubleSpeedThreshold = 64;
+    }
+
+    private void OnVideoModeChanged(object? sender, VideoModeChangedEventArgs e) {
+        Reset();
+    }
+
+    private void CallUserSubRoutine() {
+        SaveRegisters();
+        // Set mouse info
+        MouseStatus status = GetCurrentMouseStatus();
+        _state.AX = (ushort)_mouseDevice.LastTrigger;
+        _state.BX = (ushort)status.ButtonFlags;
+        _state.CX = (ushort)status.X;
+        _state.DX = (ushort)status.Y;
+        _state.SI = (ushort)GetDeltaXMickeys();
+        _state.DI = (ushort)GetDeltaYMickeys();
+
+        if (_logger.IsEnabled(LogEventLevel.Verbose)) {
+            _logger.Verbose("{ClassName} {MethodName}: calling {Segment:X4}:{Offset:X4} with AX={AX:X4}, BX={BX:X4}, CX={CX:X4}, DX={DX:X4}, SI={SI:X4}, DI={DI:X4}",
+                nameof(MouseDriver), nameof(CallUserSubRoutine), _userCallback.Segment, _userCallback.Offset, _state.AX, _state.BX, _state.CX, _state.DX, _state.SI, _state.DI);
+        }
+
+        // We're going to call the user specific subroutine, and then return to a special callback that will restore the registers.
+        const int int90HandlerVector = 0x90;
+        ushort callAddressSegment = _memory.GetUint16(4 * int90HandlerVector + 2);
+        ushort callAddressOffset = _memory.GetUint16(4 * int90HandlerVector);
+
+        _cpu.FarCall(callAddressSegment, callAddressOffset, _userCallback.Segment, _userCallback.Offset);
```

Comment:
In spice86dumpExecutionFlow.json it already shows up as a call from F000:002C (`interrupt_handler_0x74_F000_002C_F002C`) to 1834:492E (`unknown_1834_492E_1CC6E`)

What exactly is going wrong in the control flow analysis and code generator with the current solution?

Reply:
Current solution is based on ghidra and will not generate a call because instruction CALLBACK doesnt exist in ghidra (actually we convert the callbacks to INT as a hack)

Next solution could work with this but I would prefer to have an explicit CALLF so that control flow is explicit and we dont have to handle control flow for CALLBACK instructions (and in this case it is not only a jump, it is a call which implies stack modification).

Reply:
Ok. I'm not sure how to do that. Could you make the required modifications to this PR?